buf/registry/module/v1beta1/commit_service.proto

```diff
@@ -310,14 +310,28 @@ message GetBlobsResponse {
 }
 
 message GetMissingBlobDigestsRequest {
-  // The digests to see if we have Blobs for.
-  repeated buf.registry.storage.v1beta1.Digest digests = 1 [
+  message ModuleBlobDigests {
+    // The reference to the Module that the digests belong to.
+    ModuleRef module = 1;
+    // The digests to see if we have Blobs for.
+    repeated buf.registry.storage.v1beta1.Digest blob_digests = 2 [
+      (buf.validate.field).repeated.min_items = 1,
+      (buf.validate.field).repeated.max_items = 250
+    ];
+  }
+  repeated ModuleBlobDigests module_blob_digests = 1 [
     (buf.validate.field).repeated.min_items = 1,
     (buf.validate.field).repeated.max_items = 250
   ];
 }
 
 message GetMissingBlobDigestsResponse {
+  message MissingBlobDigest {
+    // The reference to the Module that the missing digests are for.
+    ModuleRef module = 1;
```

Comment:
@bufdev what type would I use here? `ModuleRef` is marked a `request_only`.

Is this canonical?
```proto
  string module_id = 6 [
    (buf.validate.field).required = true,
    (buf.validate.field).string.uuid = true
  ];
```
If so, that's incongruent with `ModuleRef`, which may not be an ID. Ideally the caller doesn't have to make other calls to fetch IDs for the modules (if that's the case, why take in `ModuleRef` at all).

Reply:
It should be `Module module = 1` on the response. Good catch.

Reply:
and required